### ISSUE TITLE
All Schedules List

### ISSUE THREAD

**Issue by mabashian** [ansible/awx#5857]
##### ISSUE TYPE
 - Feature Idea

##### SUMMARY
This issue tracks work for building the list shown on `/#/schedules`.  It should be shown when the user clicks on this sidenav item:

<img width="169" alt="Screen Shot 2020-02-10 at 1 12 50 PM" src="https://user-images.githubusercontent.com/9889020/74177124-1e984880-4c07-11ea-8a78-23ae289bc8a7.png">

The list is populated by interacting with `/api/v2/schedules/`. 

The list should have 5 columns:

1) Name (shows the name of the schedule and should link to the schedule within the context of whatever resource it belongs to).
    - For example: If you click on a scheduled project sync's name it should take you to `/#/projects/:project_id/schedules/:schedule_id`.
2) Type (Management Job, Project Sync, etc)
3) Next Run (Shows the timestamp of the next run)
4) On/Off Toggle (Allows the user to toggle the schedule on and off - rbac protected)
5) Edit Button (See column 1.  This should link to the edit schedule within the context of whatever resource it belongs to - also rbac protected).

Deletion is also supported given that rbac allows it.

Sortable keys:

1) Name (name)
2) Type (unified_job_template__polymorphic_ctype__model)
3) Next Run (next_run)

Searchable keys:

Should probably match the sortable keys for now - use your discretion.

Here's a link to the mockup: https://tower-mockups.testing.ansible.com/patternfly/schedules/schedules/

**Comment by mabashian** [comment]
This issue and https://github.com/ansible/awx/issues/4959 should probably be worked on by the same individual.

**Comment by dsesami** [comment]
this, as all lists, suffers from the wrapping bug with DataListCells here: https://github.com/ansible/awx/issues/6351, but that is non-blocking. 

Links works, sorts work. 

Two things 

1. When you click a hyperlink, it offers to take you "Back to Schedules", but it takes you to the schedules tab for that Job Template, as opposed to the overall schedules list. Should this be a "smart" back-link that distinguishes the two? Seems like it should be.
2. I can only search by Name. @jlmitch5 @mabashian should there be other searchable keys here?

**Comment by mabashian** [comment]
1. _Maybe_.  I definitely see what you're saying.  We could definitely try to determine whether the user came from the all schedules list or not.  I think this is worth a follow-on issue.
2. Yea there probably are other fields that a user might want to search on.  Maybe enabled/disabled, searching on the various dates (created, edited, start datetime, etc).  It may also be helpful to give them a quick filter to show schedules that will never run again (end datetime is < now).

**Comment by jlmitch5** [comment]
👍 those all sound like great search keys @mabashian 

the date ones may? be blocked by https://github.com/ansible/awx/issues/6214 currently the search does not implement the date input element (but could be made as an enhancement).  that maybe should be a follow-up issue?  up to you

**Comment by dsesami** [comment]
This is looking good, and the follow-up issues have been created. Closing it out.
